.test-infra/jenkins/job_PerformanceTests_SingleStoreIO.groovy

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import CommonJobProperties as common
+import Kubernetes
+import InfluxDBCredentialsHelper
+
+String jobName = "beam_PerformanceTests_SingleStoreIO"
+
+void waitForPodWithLabel(job, Kubernetes k8s, String label) {
+  job.steps {
+    shell("${k8s.KUBERNETES_DIR}/singlestore/wait-for-pod-with-label.sh ${label} 600")
+  }
+}
+
+void waitFor(job, Kubernetes k8s, String resource) {
+  job.steps {
+    shell("${k8s.KUBERNETES_DIR}/singlestore/wait-for.sh ${resource} 600")
+  }
+}
+
+job(jobName) {
+  common.setTopLevelMainJobProperties(delegate)
+  common.setAutoJob(delegate,'H H/6 * * *')
+  common.enablePhraseTriggeringFromPullRequest(
+      delegate,
+      'Java SingleStoreIO Performance Test',
+      'Run Java SingleStoreIO Performance Test')
+  InfluxDBCredentialsHelper.useCredentials(delegate)
+
+
+  String namespace = common.getKubernetesNamespace(jobName)
+  String kubeconfigPath = common.getKubeconfigLocationForNamespace(namespace)
+  Kubernetes k8s = Kubernetes.create(delegate, kubeconfigPath, namespace)
+
+  k8s.apply(common.makePathAbsolute("src/.test-infra/kubernetes/singlestore/sdb-rbac.yaml"))
+  k8s.apply(common.makePathAbsolute("src/.test-infra/kubernetes/singlestore/sdb-cluster-crd.yaml"))
+  k8s.apply(common.makePathAbsolute("src/.test-infra/kubernetes/singlestore/sdb-operator.yaml"))
+  waitForPodWithLabel(delegate, k8s, "sdb-operator")
+
+  k8s.apply(common.makePathAbsolute("src/.test-infra/kubernetes/singlestore/sdb-cluster.yaml"))
+  waitFor(delegate, k8s, "memsqlclusters.memsql.com")
+
+  String singlestoreHostName = "LOAD_BALANCER_IP"
+  k8s.loadBalancerIP("svc-sdb-cluster-ddl", singlestoreHostName)
+
+  Map pipelineOptions = [
+    tempRoot                  : 'gs://temp-storage-for-perf-tests',
+    project                   : 'apache-beam-testing',
+    runner                    : 'DataflowRunner',
+    singleStoreServerName     : "\$${singlestoreHostName}",
+    singleStoreUsername : "admin",
+    singleStorePassword : "secretpass",
+    singleStorePort: "3306",
+    numberOfRecords: "100000",
```

Comment:
Looks like the performance test is passing. The Write then Read steps finishing within seconds. To get a more meaningful metric, ideally the pipeline runs several minutes. How about setting  numberOfRecords to 10000000?

Reply:
Sorry I meant 10M records (100 times of "100000").

The metrics shown in https://ci-beam.apache.org/view/PerformanceTests/job/beam_PerformanceTests_SingleStoreIO/6/console
```
13:54:32     Load test results for test (ID): 403b18ab-b3dc-475a-b4d9-0a77866159c8 and timestamp: 2022-11-15T18:54:32.477000000Z:
13:54:32                      Metric:                    Value:
13:54:32                   write_time                     0.628
13:54:32     Load test results for test (ID): 403b18ab-b3dc-475a-b4d9-0a77866159c8 and timestamp: 2022-11-15T18:54:32.477000000Z:
13:54:33                      Metric:                    Value:
13:54:33                    read_time                     4.433
13:54:33     Load test results for test (ID): 403b18ab-b3dc-475a-b4d9-0a77866159c8 and timestamp: 2022-11-15T18:54:32.477000000Z:
13:54:33                      Metric:                    Value:
13:54:33     read_with_partitions_time                      1.87
```
If performance is linear 1M record still costs seconds to execute write.

SingleStoreDB is pretty performant!

Reply:
Wow, it is really pretty fast :)
Increased the number of rows to 10M.

Reply:
ah sorry. The test row pre-defined some hash for validation. 10000000 does not work but 5000000 will do:
```
  private static final ImmutableMap<Integer, String> EXPECTED_HASHES =
      ImmutableMap.of(
          1000, "7d94d63a41164be058a9680002914358",
          100_000, "c7cbddb319209e200f1c5eebef8fe960",
          600_000, "e2add2f680de9024e9bc46cd3912545e",
          5_000_000, "c44f8a5648cd9207c9c6f77395a998dc");
```